Akka-http

igreenfield opened the pull request:
Scala 2.12
Migration spray to akka-http
Move to Spark 2.4.0

**Pull Request checklist**

- [ ] The commit(s) message(s) follows the contribution [guidelines](doc/contribution-guidelines.md#commit-message-format) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**New behavior :**

**Other information**:
Python is failing so it commented out
<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/spark-jobserver/spark-jobserver/1131)
<!-- Reviewable:end -->

velvia commented:
Thanks for the PR!   This is pretty awesome.

On Wed, Nov 14, 2018 at 5:19 AM Izek Greenfield <notifications@github.com>
wrote:

> Scala 2.12
> Migration spray to akka-http
> Move to Spark 2.4.0
>
> Python is failing so it commented out
>
> *Pull Request checklist*
>
>    - The commit(s) message(s) follows the contribution guidelines
>    <http://doc/contribution-guidelines.md#commit-message-format> ?
>    - Tests for the changes have been added (for bug fixes / features) ?
>    - Docs have been added / updated (for bug fixes / features) ?
>
> *Current behavior :* (link exiting issues here :
> https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests
> )
>
> *New behavior :*
>
> *BREAKING CHANGES*
>
> If this PR contains a breaking change, please describe the impact and
> migration
> path for existing applications.
> If not please remove this section.
>
> *Other information*:
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/spark-jobserver/spark-jobserver/pull/1131
> Commit Summary
>
>    - Move to scala 2.12
>
> File Changes
>
>    - *M*
>    akka-app/src/main/scala/spark/jobserver/common/akka/AkkaTestUtils.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-0>
>    (3)
>    - *M*
>    akka-app/src/main/scala/spark/jobserver/common/akka/actor/Reaper.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-1>
>    (2)
>    - *M*
>    akka-app/src/main/scala/spark/jobserver/common/akka/web/CommonRoutes.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-2>
>    (9)
>    - *M*
>    akka-app/src/main/scala/spark/jobserver/common/akka/web/WebService.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-3>
>    (28)
>    - *M*
>    akka-app/src/test/scala/spark/jobserver/common/akka/actor/ReaperSpec.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-4>
>    (2)
>    - *M*
>    akka-app/src/test/scala/spark/jobserver/common/akka/web/CommonRoutesSpec.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-5>
>    (48)
>    - *M* build.sbt
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-6>
>    (38)
>    - *M*
>    job-server-extras/src/main/scala/spark/jobserver/KMeansExample.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-7>
>    (4)
>    - *M*
>    job-server-extras/src/main/scala/spark/jobserver/NamedObjectsTestJob.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-8>
>    (4)
>    - *M*
>    job-server-extras/src/main/scala/spark/jobserver/SparkSessionJob.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-9>
>    (2)
>    - *M*
>    job-server-python/src/test/scala/spark/jobserver/python/SubprocessSpec.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-10>
>    (18)
>    - *M*
>    job-server/src/main/scala/db/h2/migration/V0_7_3/V0_7_3__Migrate_Blobs.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-11>
>    (2)
>    - *M*
>    job-server/src/main/scala/db/h2/migration/V0_7_5/V0_7_5__Migrate_Blobs.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-12>
>    (2)
>    - *M*
>    job-server/src/main/scala/db/h2/migration/V0_7_8/V0_7_8__add_contextId_and_state_to_jobs.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-13>
>    (2)
>    - *M* job-server/src/main/scala/db/migration/V0_7_5/Migration.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-14>
>    (4)
>    - *M* job-server/src/main/scala/db/migration/V0_7_8/Migration.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-15>
>    (4)
>    - *M*
>    job-server/src/main/scala/db/mysql/migration/V0_7_5/V0_7_5__Migrate_Blobs.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-16>
>    (2)
>    - *M*
>    job-server/src/main/scala/db/mysql/migration/V0_7_8/V0_7_8__add_contextId_and_state_to_jobs.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-17>
>    (2)
>    - *M*
>    job-server/src/main/scala/db/postgresql/migration/V0_7_3/V0_7_3__Migrate_Blobs.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-18>
>    (2)
>    - *M*
>    job-server/src/main/scala/db/postgresql/migration/V0_7_5/V0_7_5__Migrate_Blobs.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-19>
>    (2)
>    - *M*
>    job-server/src/main/scala/db/postgresql/migration/V0_7_8/V0_7_8__add_contextId_and_state_to_jobs.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-20>
>    (2)
>    - *M*
>    job-server/src/main/scala/spark/jobserver/AkkaClusterSupervisorActor.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-21>
>    (12)
>    - *M*
>    job-server/src/main/scala/spark/jobserver/ChunkEncodedStreamingSupport.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-22>
>    (36)
>    - *M*
>    job-server/src/main/scala/spark/jobserver/ChunkEncodingActor.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-23>
>    (96)
>    - *M* job-server/src/main/scala/spark/jobserver/CorsSupport.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-24>
>    (86)
>    - *M* job-server/src/main/scala/spark/jobserver/JobInfoActor.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-25>
>    (10)
>    - *M* job-server/src/main/scala/spark/jobserver/JobManager.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-26>
>    (24)
>    - *M* job-server/src/main/scala/spark/jobserver/JobServer.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-27>
>    (2)
>    - *M*
>    job-server/src/main/scala/spark/jobserver/JobServerNamedObjects.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-28>
>    (67)
>    - *M* job-server/src/main/scala/spark/jobserver/WebApi.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-29>
>    (678)
>    - *M* job-server/src/main/scala/spark/jobserver/auth/AuthInfo.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-30>
>    (2)
>    - *M*
>    job-server/src/main/scala/spark/jobserver/auth/SJSAuthenticator.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-31>
>    (110)
>    - *M*
>    job-server/src/main/scala/spark/jobserver/cassandra/Cassandra.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-32>
>    (11)
>    - *M*
>    job-server/src/main/scala/spark/jobserver/io/JobCassandraDAO.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-33>
>    (2)
>    - *M* job-server/src/main/scala/spark/jobserver/io/JobDAO.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-34>
>    (22)
>    - *M* job-server/src/main/scala/spark/jobserver/io/JobSqlDAO.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-35>
>    (7)
>    - *M* job-server/src/main/scala/spark/jobserver/routes/DataRoutes.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-36>
>    (101)
>    - *M*
>    job-server/src/main/scala/spark/jobserver/slick/unmanaged/UnmanagedDatabase.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-37>
>    (1)
>    - *M* job-server/src/main/scala/spark/jobserver/util/ForcefulKill.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-38>
>    (63)
>    - *A* job-server/src/test/scala/spark/jobserver/auth/AuthMagnet.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-39>
>    (32)
>    - *M*
>    job-server/src/test/scala/spark/jobserver/auth/WebApiWithAuthenticationSpec.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-40>
>    (233)
>    - *M* project/Dependencies.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-41>
>    (21)
>    - *M* project/Versions.scala
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-42>
>    (14)
>    - *M* scalastyle-config.xml
>    <https://github.com/spark-jobserver/spark-jobserver/pull/1131/files#diff-43>
>    (2)
>
> Patch Links:
>
>    - https://github.com/spark-jobserver/spark-jobserver/pull/1131.patch
>    - https://github.com/spark-jobserver/spark-jobserver/pull/1131.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/spark-jobserver/spark-jobserver/pull/1131>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABA32wtFf6klZ_H286ptoMdCe0ypuAXBks5uvBhSgaJpZM4YduY3>
> .
>


-- 
If you are free, you need to free somebody else.
If you have some power, then your job is to empower somebody else.
   --- Toni Morrison

igreenfield commented:
Hi @velvia 
1. I Fixed all the compilation error. 
2, I do not familiar with the python staff so I will need help with that.
3. I cleaned up all the comments.

igreenfield commented:
Hi @valan4ik.
Could you help me understand why the build is falling?
Thanks Izek

igreenfield commented:
test this please

valan4ik commented:
Hi @igreenfield,
I was trying to review your changes and run the tests/build locally yesterday. I have to say, that the change is going a bit out of control: 93 files and 2 700 lines of code :)
It's very hard to review at the moment and to understand what is changed and why.

Could you maybe take some time to separate the changes and make several clean commits/pull requests for scala/akka-http/refactoring/..?
Also check out [contribution documentation](https://github.com/spark-jobserver/spark-jobserver/blob/master/doc/contribution-guidelines.md) for guidelines.

P.S. I also see, that Jenkins tests are running now (although seem to have some errors, but still huge step forward) 👍

igreenfield commented:
Hi @valan4ik,
1. there is a problem with timing in the tests, I see that tests the I do not touch any code in failing and if I locally increase timeout the pass.
2. Most of Python test are failing I need help with it. I don't know enough pyspark...
3. I will do refactoring to the commits.

noorul commented:
@spark-jobserver/committers Are we targeting this for 0.9.0 ?

bsikander commented:
I am not sure. From my perspective, it is too risky. Too many changes at the last moment.

It would be great if we can split this change into Spark 2.4 and Akka Http.
Spark 2.4 can be targeted for 0.9.0.

igreenfield commented:
python test failed on spark 2.4.0: https://issues.apache.org/jira/browse/SPARK-26019

velvia commented:
Izek, first I really appreciate the effort here.  Akka HTTP is definitely a
big feature going forward.

I do agree with Behroz and others that this is a bit huge of a change to
swallow at the same time.   My recommendation is also to first have a Spark
2.4 upgrade PR, then we do a release, then we can consume this PR which
will become a bit smaller by then.   WDYT?


On Mon, Nov 26, 2018 at 11:42 PM Izek Greenfield <notifications@github.com>
wrote:

> python test failed on spark 2.4.0:
> https://issues.apache.org/jira/browse/SPARK-26019
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spark-jobserver/spark-jobserver/pull/1131#issuecomment-441958475>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABA32xMjfEPB1xmsMDodPF16OF1I3HG7ks5uzOzTgaJpZM4YduY3>
> .
>


-- 
If you are free, you need to free somebody else.
If you have some power, then your job is to empower somebody else.
   --- Toni Morrison

igreenfield commented:
You can't release with 2.4 when python does not work

bsikander commented:
If 2.4 has an issue with Python, then should we fallback to 2.3.2?

bsikander commented:
@igreenfield could you extract the upgrade change and make it fallback to 2.3.2 instead of 2.4. This way we can merge the 2.3.2 change first and your Akka Http change can go in next.

bsikander commented:
@igreenfield Why do you remove "net.jpountz.lz4" dependency?

valan4ik commented:
Closing this PR due to inactivity and because now https://github.com/spark-jobserver/spark-jobserver/pull/1332 was opened for Akka HTTP.
Also support for Spark 2.4 was already added.
